scripts/install/common/validation.test.sh

```
@@ -0,0 +1,311 @@
+##############################################################################
+
+# Check if jq is available before running parse_package_json tests
+if command -v jq &> /dev/null; then
+    test_parse_package_json_functional() {
+        local test_name="$1"
+        local expected_result="$2"
+        
+        test_start "parse_package_json: $test_name"
+        
+        # Create temporary directory and package.json for testing
+        local temp_dir=$(mktemp -d)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)